### PR TITLE
fix(billing): Handle errored out metrics project request

### PR DESCRIPTION
**Pull request by AdityaHegde**
- [x] Handle the metrics project credentials request erroring out.
- [x] Use displayName in plan to keep the mapping in backend.
- [x] Show loading spinner instead of org selector during deploy.
- [x] Hide payment module for trial plans.

### CODE REVIEW

File: web-admin/src/features/billing/plans/PlanQuotas.svelte [hunk 1]
```diff
@@ -23,7 +23,7 @@
       : "Unlimited";
 
   $: usageMetrics = getOrganizationUsageMetrics(organization);
-  $: total = $usageMetrics?.data?.reduce((s, m) => s + m.size, 0) ?? 0;
+  $: totalOrgSize = $usageMetrics?.data?.reduce((s, m) => s + m.size, 0) ?? 0;
```
Comment (ericpgreen2) [inline web-admin/src/features/billing/plans/PlanQuotas.svelte:26]: Nit: I read `totalOrgSize` as "number of people in the organization". `totalOrgUsage`?

File: web-admin/src/features/billing/plans/utils.ts [hunk 2]
```diff
@@ -4,18 +4,18 @@ import { DateTime } from "luxon";
...
 export function formatDataSizeQuota(
-  totalSizeInBytes: number,
+  sizeInBytes: number,
   storageLimitBytesPerDeployment: string,
 ): string {
   const maxSize = Number(storageLimitBytesPerDeployment);
   if (Number.isNaN(maxSize) || storageLimitBytesPerDeployment === "-1")
     return "";
-  const formattedTotal = formatMemorySize(totalSizeInBytes);
+  const formattedTotal = formatMemorySize(sizeInBytes);
```
Comment (ericpgreen2) [inline web-admin/src/features/billing/plans/utils.ts:13]: Nit: "total" is hard to understand in this function. What if we used the terms "usage" (the numerator) and "quota" (the denominator)? The function name could be `formatUsageOfQuota`

File: web-admin/src/features/errors/error-utils.ts [hunk 3]
```diff
@@ -139,6 +142,12 @@ export function createGlobalErrorCallback(queryClient: QueryClient) {
       return;
     }
 
+    // Handle case when usage metrics project is not available
+    // This is always the case on non-prod
+    if (isMetricsProjectQuery(query)) {
```
Comment (ericpgreen2) [inline web-admin/src/features/errors/error-utils.ts:147]: It's not clear to me why this function would return false outside of production?

**Review by ericpgreen2 — APPROVED** [review]

File: web-admin/src/features/errors/error-utils.ts (hunk 3 above)
Comment (AdityaHegde) [reply web-admin/src/features/errors/error-utils.ts]: I have removed that comment. It could be down in prod as well.

Comment (ericpgreen2) [reply web-admin/src/features/errors/error-utils.ts]: Oh, this is a custom API, so if the Rill API resource has not successfully reconciled, the API path will be undefined or something?

Comment (AdityaHegde) [reply web-admin/src/features/errors/error-utils.ts]: Ya that could be the case on prod. Right now, on non-prod it is not preset at all.

**Pull request merged**